packages/grid/data-grid/package.json

```
@@ -45,8 +45,8 @@
     "reselect": "^4.0.0"
   },
   "peerDependencies": {
-    "@material-ui/core": "^4.9.12 || ^5.0.0-beta.0",
-    "@material-ui/styles": "^4.9.10 || ^5.0.0-beta.0",
+    "@material-ui/core": "^4.12.0 || ^5.0.0-beta.0",
+    "@material-ui/styles": "^4.12.0 || ^5.0.0-beta.0",
```

The lowest @material-ui/styles required by the core v4 is 4.12.0.

https://github.com/mui-org/material-ui/blob/53a1655143aa4ec36c29a6063ccdf89c48a74bfd/packages/material-ui/package.json#L53

```suggestion
    "@material-ui/styles": "^4.11.4 || ^5.0.0-beta.0",
```